File: 0544_output_contest_matches/Solution.scala
// LeetCode 0544 - Output Contest Matches
// https://leetcode.com/problems/output-contest-matches/

object Solution {
  def findContestMatch(n: Int): String = {
    var teams = (1 to n).map(_.toString).toList
    while (teams.length > 1) {
      val nextRound = teams.indices
        .filter(_ < teams.length / 2)
        .map(i => s"(${teams(i)},${teams(teams.length - 1 - i)})")
        .toList
      teams = nextRound
    }
    teams.head
  }
}

File: 0544_output_contest_matches/Solution.java
// LeetCode 0544 - Output Contest Matches
// https://leetcode.com/problems/output-contest-matches/

import java.util.ArrayList;
import java.util.List;

class Solution {
    public String findContestMatch(int n) {
        List<String> teams = new ArrayList<>();
        for (int i = 1; i <= n; i++) {
            teams.add(String.valueOf(i));
        }

        while (teams.size() > 1) {
            List<String> nextRound = new ArrayList<>();
            for (int i = 0; i < teams.size() / 2; i++) {
                nextRound.add("(" + teams.get(i) + "," + teams.get(teams.size() - 1 - i) + ")");
            }
            teams = nextRound;
        }

        return teams.get(0);
    }
}
